java/server/src/org/openqa/selenium/grid/data/NewSessionResponse.java

```diff
@@ -0,0 +1,91 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.grid.data;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.json.JsonInput;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.UUID;
+
+public class NewSessionResponse {
+
+  private final Session session;
+  private final byte[] downstreamEncodedResponse;
+  private final UUID requestId;
+
+  public NewSessionResponse(Session session, byte[] downstreamEncodedResponse, UUID requestId) {
```

Comment:
Nit: the `requestId` is the primary identifier. Feel free to make it the first parameter